File: app/src/main/java/droiddevelopers254/droidconke/views/fragments/ScheduleFragment.java
package droiddevelopers254.droidconke.views.fragments;

import android.os.Bundle;
import androidx.annotation.NonNull;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;
import com.google.android.material.tabs.TabLayout;

import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.List;

import droiddevelopers254.droidconke.HomeActivity;
import droiddevelopers254.droidconke.R;

public class ScheduleFragment extends Fragment {
    private TabLayout tabLayout;
    private ViewPager viewPager;



    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_schedule, container, false);
        getActivity().setTitle("Schedule");

        viewPager = view.findViewById(R.id.viewpager);
        setupViewPager(viewPager);

        tabLayout =view.findViewById(R.id.tabs);
        tabLayout.setupWithViewPager(viewPager);

        int currentItem = viewPager.getCurrentItem();
        if (currentItem ==2){
            HomeActivity.fabVisible=false;
            Toast.makeText(getActivity(),"true",Toast.LENGTH_SHORT).show();
        }
        return view;
    }
     private void setupViewPager(ViewPager viewPager) {
        ViewPagerAdapter adapter = new ViewPagerAdapter(getChildFragmentManager());
        adapter.addFragment(new DayOneFragment(),getString(R.string.day_one_label));
        adapter.addFragment(new DayTwoFragment(),getString(R.string.day_two_label));
        adapter.addFragment(new AgendaFragment(),"Agenda");
        viewPager.setAdapter(adapter);
    }
    class ViewPagerAdapter extends FragmentPagerAdapter {
        private final List<Fragment> mFragmentList = new ArrayList<>();
        private final List<String> mFragmentTitleList = new ArrayList<>();

        public ViewPagerAdapter(FragmentManager manager) {
            super(manager);
        }

        @Override
        public Fragment getItem(int position) {
            return mFragmentList.get(position);
        }

        @Override
        public int getCount() {
            return mFragmentList.size();
        }

        public void addFragment(Fragment fragment, String title) {
            mFragmentList.add(fragment);
            mFragmentTitleList.add(title);
        }

        @Override
        public CharSequence getPageTitle(int position) {
            return mFragmentTitleList.get(position);
        }
    }


}

File: app/src/main/java/droiddevelopers254/droidconke/views/fragments/ScheduleFragment.kt
package droiddevelopers254.droidconke.views.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.ViewPager
import droiddevelopers254.droidconke.HomeActivity
import droiddevelopers254.droidconke.R
import kotlinx.android.synthetic.main.fragment_schedule.view.*
import java.util.*

class ScheduleFragment : Fragment() {


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_schedule, container, false)
        activity?.title = "Schedule"

        val viewPager = view.viewpager
        val tabLayout = view.tabs

        setupViewPager(viewPager)
        tabLayout.setupWithViewPager(viewPager)

        val currentItem = viewPager.currentItem
        when (currentItem) {
            2 -> {
                HomeActivity.fabVisible = false
            }
        }
        return view
    }

    private fun setupViewPager(viewPager: ViewPager) {
        val adapter = ViewPagerAdapter(childFragmentManager)
        adapter.addFragment(DayOneFragment(), getString(R.string.day_one_label))
        adapter.addFragment(DayTwoFragment(), getString(R.string.day_two_label))
        adapter.addFragment(AgendaFragment(), "Agenda")
        viewPager.adapter = adapter
    }

    inner class ViewPagerAdapter(manager: FragmentManager) : FragmentPagerAdapter(manager) {
        private val mFragmentList = ArrayList<Fragment>()
        private val mFragmentTitleList = ArrayList<String>()

        override fun getItem(position: Int): Fragment {
            return mFragmentList[position]
        }

        override fun getCount(): Int {
            return mFragmentList.size
        }

        fun addFragment(fragment: Fragment, title: String) {
            mFragmentList.add(fragment)
            mFragmentTitleList.add(title)
        }

        override fun getPageTitle(position: Int): CharSequence? {
            return mFragmentTitleList[position]
        }
    }


}
